// jquery-jsonrpcclient/test/unit.conf.js
// Karma configuration
// Generated on Fri Feb 07 2014 08:31:06 GMT+0100 (CET)

module.exports = function(config) {
  config.set({

    // base path, that will be used to resolve files and exclude
    basePath: '..',


    // frameworks to use
    frameworks: ['mocha','chai-sinon'],


    // list of files / patterns to load in the browser
    files: [
      'node_modules/jquery/dist/jquery.js',
      'jquery.jsonrpcclient.js',
      'test/unit/**/*.js'
    ],


    // list of files to exclude
    exclude: [

    ],


    // test results reporter to use
    // possible values: 'dots', 'progress', 'junit', 'growl', 'coverage'
    reporters: ['progress','coverage','growler'],

    preprocessors: {
      // source files, that you wanna generate coverage for
      // do not include tests or libraries
      // (these files will be instrumented by Istanbul)
      'jquery.jsonrpcclient.js': ['coverage']
    },

    // optionally, configure the reporter
    coverageReporter: {
      type : 'lcov', dir : 'coverage/'
    },

    // web server port
    port: 9876,


    // enable / disable colors in the output (reporters and logs)
    colors: true,


    // level of logging
    // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
    logLevel: config.LOG_INFO,


    // enable / disable watching file and executing tests whenever any file changes
    autoWatch: true,


    // global config of your BrowserStack account
      browserStack: {
          username: 'fredrikliljegren1',
          accessKey: 'SiVgacQ7odBDbU8xyBpQ'
      },
    // define browsers
      customLaunchers: {
          bs_firefox_mac: {
              base: 'BrowserStack',
              browser: 'firefox',
              browser_version: '21.0',
              os: 'OS X',
              os_version: 'Mountain Lion'
          },
          bs_iphone5: {
              base: 'BrowserStack',
              device: 'iPhone 5',
              os: 'ios',
              os_version: '6.0'
          },
          bs_ie9: {
              base: 'BrowserStack',
              browser: 'ie',
              browser_version: '9.0',
              os: 'Windows',
              os_version: '7'
          },
          bs_ie10: {
              base: 'BrowserStack',
              browser: 'ie',
              browser_version: '10.0',
              os: 'Windows',
              os_version: '8'
          }
      },


    // Start these browsers, currently available:
    // - Chrome
    // - ChromeCanary
    // - Firefox
    // - Opera (has to be installed with `npm install karma-opera-launcher`)
    // - Safari (only Mac; has to be installed with `npm install karma-safari-launcher`)
    // - PhantomJS
    // - IE (only Windows; has to be installed with `npm install karma-ie-launcher`)
    browsers: ['bs_ie9', 'bs_ie10', 'PhantomJS'],

    // If browser does not capture in given timeout [ms], kill it
    captureTimeout: 120000,


    // Continuous Integration mode
    // if true, it capture browsers, run tests and exit
    singleRun: false
  });
};
